### PR TITLE
Quick hack to make the table memory backend sparse

### PR DESCRIPTION
**Pull request by skyem123**
This may not even work, but it's worth a shot. I'll fix it if it breaks.

### CODE REVIEW

File: memlib.lua [hunk 1]
```diff
@@ -113,11 +113,32 @@ function _M.backend.table(memsz, prealloc)
...
+		end,
+		__nextindex = function(t, i, v)
+			if v == 0 then
+				v = nil
+			end
+			memory[i] = v
+		end
+	}
+	
+	setmetatable({}, mt_tbackend)
```
Comment (vifino) [inline memlib.lua]: You set the metatable of a new table, which isn't ever going to be used, to a value evaluating to nil.

File: memlib.lua [hunk 2]
```diff
@@ -113,11 +113,32 @@ function _M.backend.table(memsz, prealloc)
...
+	local meta = {
+		__index = function(t, i)
+			local v = memory[i]
+			if v == nil then
+				return 0
+			else
+				return v
+			end
+		end,
+		__nextindex = function(t, i, v)
```
Comment (vifino) [inline memlib.lua]: __nextindex -> __newindex

**Review by vifino — CHANGES_REQUESTED** [review]

**Comment by vifino** [comment]
You should probably also move the whole metatable block out of the function and use the t variable instead of memory.
Also, rawset is very useful.

Apart from that, it is probably better to do such changes in the get/set functions, not via a metatable.

File: memlib.lua (hunk 1 above)
Comment (skyem123) [reply memlib.lua]: That... was a typo.

File: memlib.lua (hunk 2 above)
Comment (skyem123) [reply memlib.lua]: oops.

**Pull request closed without merging**